indexer/feature.cpp

```diff
@@ -475,12 +474,60 @@ uint32_t FeatureType::ParseGeometry(int scale)
     }
     m_parsed.m_points = true;
   }
-  return sz;
 }
 
-uint32_t FeatureType::ParseTriangles(int scale)
+FeatureType::GeomStat FeatureType::GetOuterGeometryStats()
+{
+  ASSERT(!m_parsed.m_points, ("Geometry had been parsed already!"));
+  CHECK(m_loadInfo, ());
+  size_t const scalesCount = m_loadInfo->GetScalesCount();
+  ASSERT_LESS_OR_EQUAL(scalesCount, DataHeader::kMaxScalesCount, ("MWM has too many geometry scales!"));
+  FeatureType::GeomStat res;
+
+  auto const headerGeomType = static_cast<HeaderGeomType>(Header(m_data) & HEADER_MASK_GEOMTYPE);
+  if (headerGeomType == HeaderGeomType::Line)
+  {
+    size_t const pointsCount = m_points.size();
+    if (pointsCount < 2)
+    {
+      // Outer geometry present.
+      ASSERT_EQUAL(pointsCount, 1, ());
+
+      FeatureType::Points points;
+
+      for (size_t ind = 0; ind < scalesCount; ++ind)
+      {
+        auto const scaleOffset = m_offsets.m_pts[ind];
+        if (scaleOffset != kInvalidOffset)
+        {
+          points.clear();
+          points.emplace_back(m_points.front());
+
+          ReaderSource<FilesContainerR::TReader> src(m_loadInfo->GetGeometryReader(ind));
+          src.Skip(scaleOffset);
+
+          serial::GeometryCodingParams cp = m_loadInfo->GetGeometryCodingParams(ind);
+          cp.SetBasePoint(points[0]);
+          serial::LoadOuterPath(src, cp, points);
+
+          res.m_sizes[ind] = static_cast<uint32_t>(src.Pos() - scaleOffset);
+          res.m_elements[ind] = points.size();
```

Comment:
Can be unified with GetOuterTrianglesStats. Store in temporary variable points or modifying m_points

Reply:
It should be possible (and unifying ParseGeometry with ParseTriangles too) though there will be quite some `ifs` to handle line/area differences. I just don't have time for this now, sorry.